Improve fleet control section (#2851)
viamrobotics/docs: docs/fleet/machines/control.md

---
title: "Control Machines"

> linkTitle: "Control"

Author: ```suggestion
linkTitle: "Control Machines"
```

## Control interface in the Viam app

The **CONTROL** tab in the [Viam app](https://app.viam.com) gives you the ability to test, monitor, and operate the machines in your fleet.
The **CONTROL** tab provides a control interface for each component and service that you have configured for you machine.

For example, if you have configured a base with wheels, you can move your machine's with an arrow pad and control the base's speed by setting its power with a slider.
If you have configured a camera component, a window in the **CONTROL** tab displays the camera output.

If you use remote control in the [Viam app](https://app.viam.com) UI, all communication to the machine uses [WebRTC](https://pkg.go.dev/go.viam.com/utils@v0.0.3/rpc#hdr-Connection).
For local communication between [parts](/build/configure/parts/#machine-parts) Viam uses gRPC or WebRTC.

{{<gif webm_src="/fleet/control.webm" mp4_src="/fleet/control.mp4" alt="Using the control tab" max-width="800px">}}

### Components

For more detailed information on how to operate and test your resources, expand the relevant resource below:

> {{% expand "arm" %}}

Reviewer: I think all component names should be capitalized (probably sentence case, not title case)
```suggestion
{{% expand "Arm" %}}
```

{{< readfile "/static/include/components/test-control/arm-control.md" >}}
{{% /expand%}}

> {{% expand "base" %}}

Author: ```suggestion
{{% expand "Base" %}}
```

> {{< readfile "/static/include/components/test-control/base-control.md" >}}

Reviewer: All of these includes files link to `/fleet/machines/#control`, so those should probably be updated to link to `/fleet/machines/control/` instead, but also it's awkward to have them link anywhere in the context of this particular page.

Some of these includes files include a heading and some don't, and the headings are inconsistent (i.e. h2 for arm vs h4 for digital interrupts).

Author: Updated link. I think the top of the file stayed fairly similar to the content in as it was in the fleet page, so think that shouldn't be a problem?

Reviewer: I mean that it makes sense for the control section of each model to link to this page, but it makes much less sense for sections of this page to link to the page the user is already on. I realize we can't easily change this because we are using the includes files, so maybe leave as-is and hope users don't find it annoying. There's not a way to take the link out conditionally when the includes is included in this page is there?

Author: we'd have to essentially make it a shortcode... I'll try

Author: because of how the include works it's not straightforward. I think it can be done, I'll make a maintenance ticket for it

{{% /expand%}}

> {{% expand "board" %}}

Author: ```suggestion
{{% expand "Board" %}}
```

{{< readfile "/static/include/components/board/test-board-digital-interrupts.md" >}}
{{% /expand%}}

{{% expand "camera" %}}
{{< readfile "/static/include/components/camera-view-camera-stream.md" >}}
{{% /expand%}}

{{% expand "encoder" %}}
{{< readfile "/static/include/components/test-control/encoder-control.md" >}}
{{% /expand%}}

{{% expand "gantry" %}}
{{< readfile "/static/include/components/test-control/gantry-control.md" >}}
{{% /expand%}}

{{% expand "generic component" %}}
{{< readfile "/static/include/components/test-control/generic-control.md" >}}
{{% /expand%}}

{{% expand "gripper" %}}
{{< readfile "/static/include/components/test-control/gripper-control.md" >}}
{{% /expand%}}

{{% expand "input controller" %}}
{{< readfile "/static/include/components/test-control/input-controller-control.md" >}}
{{% /expand%}}

{{% expand "motor" %}}
{{< readfile "/static/include/components/test-control/motor-control.md" >}}
{{% /expand%}}

> {{% expand "movement sensor (gps)" %}}

Reviewer: ```suggestion
{{% expand "Movement sensor (GPS)" %}}
```

{{< readfile "/static/include/components/test-control/movement-sensor-gps-control.md" >}}
{{% /expand%}}

> {{% expand "movement sensor (imu)" %}}

Reviewer: ```suggestion
{{% expand "Movement sensor (IMU)" %}}
```

{{< readfile "/static/include/components/test-control/movement-sensor-imu-control.md" >}}
{{% /expand%}}

{{% expand "power sensor" %}}
{{< readfile "/static/include/components/test-control/power-sensor-control.md" >}}
{{% /expand%}}

{{% expand "sensor" %}}
{{< readfile "/static/include/components/test-control/sensor-control.md" >}}
{{% /expand%}}

{{% expand "servo" %}}
{{< readfile "/static/include/components/test-control/servo-control.md" >}}
{{% /expand%}}

### Services

The following services also provide control interfaces:

> - [SLAM](/mobility/slam/cartographer/#create-a-new-map): for createing a new map

Reviewer: ```suggestion
- [SLAM](/mobility/slam/cartographer/#create-a-new-map): for creating a new SLAM map and for using the motion service to move a machine on a SLAM map
```

> - [Motion](/mobility/motion/#test-the-motion-service): for moving a machine on a map

Reviewer: Is this true? Shouldn't this be 
```suggestion
- [Navigation](/mobility/navigation/#control-tab-usage): for moving a machine to waypoints on a map
```
The currently-shown image on https://docs.viam.com/mobility/motion/#test-the-motion-service shows the SLAM card in the control tab. The motion service can be used for 2D motion planning on a SLAM map, but this section of the motion page might make users think that it's not for 3D motion planning too. The motion service itself doesn't have a control tab card, so this "Test the motion service" section should probably be fixed. Might make sense to fix within this PR?

Author: https://docs-test.viam.dev/2851/mobility/motion/#test-the-motion-service seems to indicate so, but I should add navigation

The **CONTROL** tab provides a control interface for each component and service that you have configured for you machine.

For example, if you have configured a base with wheels, you can move your machine's with an arrow pad and control the base's speed by setting its power with a slider.
If you have configured a camera component, a window in the **CONTROL** tab displays the camera output.

If you use remote control in the [Viam app](https://app.viam.com) UI, all communication to the machine uses [WebRTC](https://pkg.go.dev/go.viam.com/utils@v0.0.3/rpc#hdr-Connection).
For local communication between [parts](/build/configure/parts/#machine-parts) Viam uses gRPC or WebRTC.

{{<gif webm_src="/fleet/control.webm" mp4_src="/fleet/control.mp4" alt="Using the control tab" max-width="800px">}}

### Components

For more detailed information on how to operate and test your resources, expand the relevant resource below:

{{% expand "arm" %}}
{{< readfile "/static/include/components/test-control/arm-control.md" >}}
{{% /expand%}}

{{% expand "base" %}}
{{< readfile "/static/include/components/test-control/base-control.md" >}}
{{% /expand%}}

{{% expand "board" %}}

> 

Author: ```suggestion
```
